### PR TITLE
Final Changes to US10 and US24

### PR DESCRIPTION
**Pull request by frinojais**

### CODE REVIEW

File: src/TwoBucks/UpdateProfile.java [hunk 1]
```diff
@@ -1,75 +1,96 @@
...
+     * This function will read from the text file where the
+     * User information is stored and change the user's choice
+     * of first name, last name, or email address
+     *
+     * @param args
+     * @throws Exception
+     */
+    public static void readFile(String[] args) throws Exception{
+        // creating object for User.txt where data is saved
         File txtFile = new File("User.txt");
```
Comment (LA-Gonzalez) [inline src/TwoBucks/UpdateProfile.java]: Save User already handles saving changes to .txt file

File: src/TwoBucks/UpdateProfile.java [hunk 2]
```diff
@@ -1,75 +1,96 @@
...
 
-            int rowIndex = 0;
+            int rowIndex = 0; // set the index to 0 to be the position of first name
 
             System.out.println("Please confirm your existing first name: ");
-            String oldFirstName = scnr.next();
+            String oldFirstName = input.next();
             System.out.println("Please enter your new first name: ");
-            String newFirstName = scnr.next();
+            String newFirstName = input.next();
```
Comment (LA-Gonzalez) [inline src/TwoBucks/UpdateProfile.java]: consider using setFirstName method from the User class

File: src/TwoBucks/UpdateProfile.java [hunk 3]
```diff
@@ -1,75 +1,96 @@
...
 
         }else if(option == 2){
             int rowIndex = 1;
 
             System.out.println("Please confirm your existing last name: ");
-            String oldLastName = scnr.next();
+            String oldLastName = input.next();
             System.out.println("Please enter your new last name: ");
-            String newLastName = scnr.next();
+            String newLastName = input.next();
```
Comment (LA-Gonzalez) [inline src/TwoBucks/UpdateProfile.java]: consider using setLastName method from the User class

File: src/TwoBucks/UpdateProfile.java [hunk 4]
```diff
@@ -84,24 +105,24 @@ public static void readTxtFile(String[] args) throws Exception{
...
 
         }else if(option == 3){
             int rowIndex = 2;
 
             System.out.println("Please confirm your existing email: ");
-            String oldEmail = scnr.next();
+            String oldEmail = input.next();
             System.out.println("Please enter your new email: ");
-            String newEmail = scnr.next();
+            String newEmail = input.next();
```
Comment (LA-Gonzalez) [inline src/TwoBucks/UpdateProfile.java]: consider using setEmail method from the User class

File: src/TwoBucks/UpdateProfile.java [hunk 5]
```diff
@@ -1,75 +1,96 @@
...
-    public static void readTxtFile(String[] args) throws Exception{
+    /***
+     * This function will read from the text file where the
+     * User information is stored and change the user's choice
+     * of first name, last name, or email address
+     *
+     * @param args
+     * @throws Exception
+     */
+    public static void readFile(String[] args) throws Exception{
```
Comment (LA-Gonzalez) [inline src/TwoBucks/UpdateProfile.java]: Consider changing to accept User as parameters and return User

File: src/TwoBucks/CalculateGoalsVsPerformance.java [hunk 6]
```diff
@@ -1,23 +1,36 @@
...
+     * user has been spending/saving.
+     */
+    public void PerformanceAnalysis() { // function to show statistics of spending/saving vs actual
 
-        EnterFinancialGoals goals = new EnterFinancialGoals();
-        EnterWeeklyExpense expense = new EnterWeeklyExpense();
-        EnterWeeklyIncome income = new EnterWeeklyIncome();
+        EnterFinancialGoals goals = new EnterFinancialGoals(); // creating an object to retrieve user's goals
+        EnterWeeklyExpense expense = new EnterWeeklyExpense(); //  creating object to retrieve user's expenses
+        EnterWeeklyIncome income = new EnterWeeklyIncome(); // creating object to retrieve user's income
```
Comment (LA-Gonzalez) [inline src/TwoBucks/CalculateGoalsVsPerformance.java]: User's income, goals, and expenses can be retrieved with get methods from users class.

File: src/TwoBucks/CalculateGoalsVsPerformance.java [hunk 7]
```diff
@@ -1,23 +1,36 @@
...
 
-    private static DecimalFormat df = new DecimalFormat("0.00");
+    private static DecimalFormat df = new DecimalFormat("0.00"); // to keep the values in the correct format
 
-    public void PerformanceAnalysis() {
+    /***
+     * This function will take the data from other classes to show results of how the
+     * user has been spending/saving.
+     */
+    public void PerformanceAnalysis() { // function to show statistics of spending/saving vs actual
```
Comment (LA-Gonzalez) [inline src/TwoBucks/CalculateGoalsVsPerformance.java]: Pass User as parameter in order to get current user information

File: src/TwoBucks/CalculateGoalsVsPerformance.java (hunk 6 above)
Comment (frinojais) [reply src/TwoBucks/CalculateGoalsVsPerformance.java]: Done

File: src/TwoBucks/CalculateGoalsVsPerformance.java (hunk 7 above)
Comment (frinojais) [reply src/TwoBucks/CalculateGoalsVsPerformance.java]: Done

File: src/TwoBucks/UpdateProfile.java (hunk 5 above)
Comment (frinojais) [reply src/TwoBucks/UpdateProfile.java]: Done

File: src/TwoBucks/UpdateProfile.java (hunk 4 above)
Comment (frinojais) [reply src/TwoBucks/UpdateProfile.java]: Done

File: src/TwoBucks/UpdateProfile.java (hunk 3 above)
Comment (frinojais) [reply src/TwoBucks/UpdateProfile.java]: Done

File: src/TwoBucks/UpdateProfile.java (hunk 2 above)
Comment (frinojais) [reply src/TwoBucks/UpdateProfile.java]: Done

File: src/TwoBucks/UpdateProfile.java (hunk 1 above)
Comment (frinojais) [reply src/TwoBucks/UpdateProfile.java]: Done

File: src/TwoBucks/CalculateGoalsVsPerformance.java [hunk 8]
```diff
@@ -24,42 +24,38 @@
      * This function will take the data from other classes to show results of how the
      * user has been spending/saving.
      */
```
Comment (LA-Gonzalez) [inline src/TwoBucks/CalculateGoalsVsPerformance.java]: Looks good!

**Pull request merged**